### DISCUSSION TITLE
Is there any way to discover the Daikin actual electrical power level?

### DISCUSSION THREAD

**Discussion by antwin** [revk/xESP32-Faikout#453] [Q&A]
I - like others - would like to control the Daikin output to match available solar power.
It would be very useful to know the actual electrical power consumption at any one time.
At the moment I am estimating it as 
(liquid - roomTemp) * fanrpm
However, this is obviously ver imprecise, non-linear and has a time lag.
I note that the mqtt status give a "Wh":2370100 parameter. 
Is this 'watt-hours"? How is it calculated?

**Comment by MUN0X** [comment]
Yes should be Wh.
I guess you could make a derivative sensor as a helper in home assistant. But I honestly doubt that the resolution of 100Wh  is small enough to produce a meaningful sensor.

I installed a power clamp on the mains to the daikin

**Reply by revk** [reply]
I too have a power clamp (Shelly, running Tasmota).

**Reply by Tarudro** [reply]
Can recommend this little guy:
[Shelly Plus PM Mini Gen3](https://www.shelly.com/de-at/products/product-overview/shelly-pm-mini-gen3)

PS. Would appreciate a short info on @antwin how you are controlling your Daikin to scale the power?
https://github.com/revk/ESP32-Faikin/discussions/450

**Comment by antwin** [comment]
Thanks for the info on the Shelley - looks interesting.
I'm controlling the power by setting the 'demand' parameter. This does seem to work, although there is a bit of a time lag between setting a demand and the 'liquid' termperature changing. But there is definitely a correlation.

**Comment by 44616E** [comment]
I do it currently two ways.
I use the Wh output and measure the time between changes plus a minimum time of 15 minutes or so. So at higher power draws it may be a few hundred watts hours in that 15 mins. This gives a result +/- 20% but is slow to update.

The other way which works for my units is I have a lookup table for compressor frequency (which is nonlinear) and also a table for power draw at different fan speeds) then just add the two together. This works reasonably well.

**Comment by rug1024** [comment]
I only use [Sonoff DUAL R2](https://templates.blakadder.com/sonoff_DUALR3_v2) with [Tasmota-Firmware](https://github.com/arendst/Tasmota) for Energy-Measurement.
Those values can of course be passed on via MQTT or processed on-device (script), possibilities with Tasmota are endless.

**Reply by 44616E** [reply]
@rug1024 I would like to do that but there is no room in the switch board.

**Comment by antwin** [comment]
I now have a Shelly installed, so I've been comparing its output with my crude algorithm. To my surprise, they are in reasonable agreement - if you take into account that there is a time lag, since I read the liquid temperature. 
In case you want to check, it's:
powerEst = (liquid - roomTemp) * fanrpm / 12 (in degrees C, and Watts)
